Source/Moya.swift

```diff
@@ -240,6 +315,59 @@ public extension MoyaProvider {
 
 internal extension MoyaProvider {
     
+    private func sendUpload(target: Target, request:NSURLRequest, multipartBody:[MultipartFormData], progress: Moya.ProgressBlock? = nil, completion: Moya.Completion) -> CancellableWrapper {
+        var cancellable = CancellableWrapper()
+        let plugins = self.plugins
+        
+        let multipartFormData = { (form:RequestMultipartFormData) -> Void in
+            let params = decodeHTTPBody(request.HTTPBody)
```

Comment:
This part is a little iffy to me. What happens if the parameter encoding wasn't `.URL`? Then the value in `params` here will be malformed. I'm not sure what'd be the best way to handle encoding of additional parameters in a multipart upload.

One option might be wrapping Alamofire's `ParameterEncoding` and providing our own encoding option for multipart? 

In any case, I think decoding the already-encoded value is problematic. We have access to all the raw values anyway (through `target.parameters` or better yet `self.endpoint(target).parameters`). I think we should use that directly.